xbmc/application/Application.cpp

```diff
@@ -2384,6 +2384,10 @@ bool CApplication::PlayFile(CFileItem item, const std::string& player, bool bRes
     return false;
   }
 
+  // Needed so stacks can be played properly from the main screen
+  if (URIUtils::IsStack(item.GetDynPath()))
+    item.SetPath(item.GetDynPath());
```

Comment:
Although we do this at several places (mostly for historical reasons), every manipulation of fileitem's path after construction has to be avoided, as the path is the identity of the item. Changing it can have unpredictable side effects.

The dynpath is exactly there to hold the url of what needs to be played. Instead of manipulating the path, the code that handles the playback needs to be fixed to use item's dynpath, not the path.

Reply:
Changed as suggested.